### DISCUSSION TITLE
Capacitor 3.x and community plugins

### DISCUSSION THREAD

**Discussion by daniell0gda** [ionic-team/capacitor#4011] [General]
It seems that community plugins are not being sync to android project using `npx cap sync`.
Is it each plugin responsibility to migrate or is it a bug (known?) in capacitor  ?

**Comment by imhoffd** [comment]
The team will be advising the plugin authors within [Capacitor community](https://github.com/capacitor-community) as to the changes necessary for Capacitor 3. I'm not sure why they aren't being synced, though. Could you post the output of `DEBUG='*' npx cap sync android` ?

**Comment by daniell0gda** [comment]
By Debug="*" you mean using [debug package](https://www.npmjs.com/package/debug)?

`set DEBUG=* & npx cap sync android

 capacitor:config config: {
  capacitor:config   android: {
  capacitor:config     name: 'android',
  capacitor:config     minVersion: '21',
  capacitor:config     studioPath: 'C:\\Program Files\\Android\\Android Studio\\bin\\studio64.exe',
  capacitor:config     platformDir: 'android',
  capacitor:config     platformDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android',
  capacitor:config     cordovaPluginsDir: 'capacitor-cordova-android-plugins',
  capacitor:config     cordovaPluginsDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\capacitor-cordova-android-plugins',
  capacitor:config     appDir: 'app',
  capacitor:config     appDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app',
  capacitor:config     srcDir: 'app/src',
  capacitor:config     srcDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app\\src',
  capacitor:config     srcMainDir: 'app/src/main',
  capacitor:config     srcMainDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app\\src\\main',
  capacitor:config     assetsDir: 'app/src/main/assets',
  capacitor:config     assetsDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app\\src\\main\\assets',
  capacitor:config     webDir: 'app/src/main/assets/public',
  capacitor:config     webDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app\\src\\main\\assets\\public',
  capacitor:config     resDir: 'app/src/main/res',
  capacitor:config     resDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app\\src\\main\\res',
  capacitor:config     buildOutputDir: 'app/build/outputs/apk/debug',
  capacitor:config     buildOutputDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\android\\app\\build\\outputs\\apk\\debug'
  capacitor:config   },
  capacitor:config   ios: {
  capacitor:config     name: 'ios',
  capacitor:config     minVersion: '12.0',
  capacitor:config     platformDir: 'ios',
  capacitor:config     platformDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\ios',
  capacitor:config     cordovaPluginsDir: 'capacitor-cordova-ios-plugins',
  capacitor:config     cordovaPluginsDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\ios\\capacitor-cordova-ios-plugins',
  capacitor:config     nativeProjectDir: 'App',
  capacitor:config     nativeProjectDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\ios\\App',
  capacitor:config     nativeTargetDir: 'App/App',
  capacitor:config     nativeTargetDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\ios\\App\\App',
  capacitor:config     webDir: 'App/public',
  capacitor:config     webDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\ios\\App\\public',
  capacitor:config     podPath: 'pod'
  capacitor:config   },
  capacitor:config   web: {
  capacitor:config     name: 'web',
  capacitor:config     platformDir: 'www',
  capacitor:config     platformDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\www'
  capacitor:config   },
  capacitor:config   cli: {
  capacitor:config     rootDir: 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor\\cli',
  capacitor:config     assetsDir: 'assets',
  capacitor:config     assetsDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor\\cli\\assets',
  capacitor:config     assets: { ios: [Object], android: [Object] },
  capacitor:config     package: {
  capacitor:config       _from: '@capacitor/cli@^3.0.0-alpha.11',
  capacitor:config       _id: '@capacitor/cli@3.0.0-alpha.11',
  capacitor:config       _inBundle: false,
  capacitor:config       _integrity: 'sha512-rNQVf1AJSgOwMOiZeOQDUMACxD+sFh6FFWu7asLwkMkDf4tKBK/FWRdiSrYvOkH/ERTVbiKWciD7eb2eYoq1mg==',
  capacitor:config       _location: '/@capacitor/cli',
  capacitor:config       _phantomChildren: [Object],
  capacitor:config       _requested: [Object],
  capacitor:config       _requiredBy: [Array],
  capacitor:config       _resolved: 'https://registry.npmjs.org/@capacitor/cli/-/cli-3.0.0-alpha.11.tgz',
  capacitor:config       _shasum: 'de2391d9004c2149da9c9a499a5f9b7e087e6d1f',
  capacitor:config       _spec: '@capacitor/cli@^3.0.0-alpha.11',
  capacitor:config       _where: 'C:\\private\\projekty\\tray sonoff temp',
  capacitor:config       author: [Object],
  capacitor:config       bin: [Object],
  capacitor:config       bugs: [Object],
  capacitor:config       bundleDependencies: false,
  capacitor:config       dependencies: [Object],
  capacitor:config       deprecated: false,
  capacitor:config       description: 'Capacitor: Cross-platform apps with JavaScript and the web',
  capacitor:config       devDependencies: [Object],
  capacitor:config       engines: [Object],
  capacitor:config       files: [Array],
  capacitor:config       gitHead: '0c73d09d6776bd1142b8929794b939caaedd9e71',
  capacitor:config       homepage: 'https://capacitorjs.com/',
  capacitor:config       jest: [Object],
  capacitor:config       keywords: [Array],
  capacitor:config       license: 'MIT',
  capacitor:config       main: 'dist/index.js',
  capacitor:config       name: '@capacitor/cli',
  capacitor:config       publishConfig: [Object],
  capacitor:config       repository: [Object],
  capacitor:config       scripts: [Object],
  capacitor:config       types: 'dist/declarations.d.ts',
  capacitor:config       version: '3.0.0-alpha.11'
  capacitor:config     },
  capacitor:config     os: 'windows'
  capacitor:config   },
  capacitor:config   app: {
  capacitor:config     rootDir: 'C:\\private\\projekty\\tray sonoff temp',
  capacitor:config     appId: 'com.tasmota.temperature.app',
  capacitor:config     appName: 'tasmota-temperature-gauge',
  capacitor:config     webDir: 'www',
  capacitor:config     webDirAbs: 'C:\\private\\projekty\\tray sonoff temp\\www',
  capacitor:config     package: {
  capacitor:config       name: 'tasmota-temperature-gauge',
  capacitor:config       productName: 'tempReader',
  capacitor:config       private: true,
  capacitor:config       version: '0.0.1',
  capacitor:config       description: 'Sonoff Th10 Tasmota Temp Reader',
  capacitor:config       main: 'Electron/app.js',
  capacitor:config       module: 'build/index.mjs',
  capacitor:config       es2015: 'build/esm/index.mjs',
  capacitor:config       es2017: 'build/esm/index.mjs',
  capacitor:config       'jsnext:main': 'build/esm/index.mjs',
  capacitor:config       scripts: [Object],
  capacitor:config       dependencies: [Object],
  capacitor:config       devDependencies: [Object],
  capacitor:config       repository: [Object],
  capacitor:config       publishConfig: [Object],
  capacitor:config       author: 'Ionic Team',
  capacitor:config       license: 'MIT',
  capacitor:config       bugs: [Object],
  capacitor:config       jest: [Object],
  capacitor:config       build: [Object],
  capacitor:config       config: [Object]
  capacitor:config     },
  capacitor:config     extConfigType: 'json',
  capacitor:config     extConfigName: 'capacitor.config.json',
  capacitor:config     extConfigFilePath: 'C:\\private\\projekty\\tray sonoff temp\\capacitor.config.json',
  capacitor:config     extConfig: {
  capacitor:config       appId: 'com.tasmota.temperature.app',
  capacitor:config       appName: 'tasmota-temperature-gauge',
  capacitor:config       bundledWebRuntime: false,
  capacitor:config       npmClient: 'npm',
  capacitor:config       webDir: 'www',
  capacitor:config       plugins: [Object],
  capacitor:config       android: [Object],
  capacitor:config       server: [Object]
  capacitor:config     },
  capacitor:config     bundledWebRuntime: false
  capacitor:config   }
  capacitor:config } +0ms

  √ Copying web assets from www to android\app\src\main\assets\public in 2.36s
√ Creating capacitor.config.json in android\app\src\main\assets in 2.61ms
√ copy android in 2.42s
√ Updating Android plugins in 14.86ms
[info] Found 3 Capacitor plugins for android:
       @capacitor-community/keep-awake@1.0.0
       @capacitor/app@0.1.2
       @capacitor/toast@0.2.2
  capacitor:android:update Searching 1 source files in 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor-community\\keep-awake\\android\\src\\main' by /^@(?:CapacitorPlugin|NativePlugin)[\s\S]+?class ([\w]+)/gm regex +0ms
  capacitor:android:update Searching 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor-community\\keep-awake\\android\\src\\main\\java\\com\\getcapacitor\\community\\keepawake\\KeepAwake.java' for package by /^package ([\w.]+);?$/gm regex +2ms
  capacitor:android:update 'com.getcapacitor.community.keepawake.KeepAwake' is a suitable plugin class +0ms
  capacitor:android:update Searching 1 source files in 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor\\app\\android\\src\\main' by /^@(?:CapacitorPlugin|NativePlugin)[\s\S]+?class ([\w]+)/gm regex +3ms
  capacitor:android:update Searching 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor\\app\\android\\src\\main\\java\\com\\capacitorjs\\plugins\\app\\AppPlugin.java' for package by /^package ([\w.]+);?$/gm regex +3ms
  capacitor:android:update 'com.capacitorjs.plugins.app.AppPlugin' is a suitable plugin class +0ms
  capacitor:android:update Searching 2 source files in 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor\\toast\\android\\src\\main' by /^@(?:CapacitorPlugin|NativePlugin)[\s\S]+?class ([\w]+)/gm regex +3ms
  capacitor:android:update Searching 'C:\\private\\projekty\\tray sonoff temp\\node_modules\\@capacitor\\toast\\android\\src\\main\\java\\com\\capacitorjs\\plugins\\toast\\ToastPlugin.java' for package by /^package ([\w.]+);?$/gm regex +1ms  capacitor:android:update 'com.capacitorjs.plugins.toast.ToastPlugin' is a suitable plugin class +0ms
√ update android in 106.97ms
[info] Sync finished in 2.534s
`

**Reply by imhoffd** [reply]
Yes, thanks. It looks like KeepAwake was detected. What do you mean by "not being sync to android project"? What error are you seeing?

**Reply by daniell0gda** [reply]
No errors just those from ionic-native are not detected. I've checked the same situation but in capacitor v.2.4.5 and they are detected.

Beside KeepAwake I've got: @ionic-native/autostart & @ionic-native/backlight.


    @capacitor-community/electron": "^1.3.2",

    "@capacitor-community/keep-awake": "^1.0.0",

    "@capacitor/android": "^3.0.0-alpha.11",

    "@capacitor/app": "^0.1.2",

    "@capacitor/cli": "^3.0.0-alpha.11",

    "@capacitor/core": "^3.0.0-alpha.9",

    "@capacitor/toast": "^0.2.2",

    "@ionic-native/autostart": "^5.30.0",

    "@ionic-native/backlight": "^5.30.0",

    "@ionic/core": "^5.5.2",

    "cordova-plugin-autostart": "^2.3.0",

    "cordova-plugin-backlight": "0.0.1",

**Reply by daniell0gda** [reply]
yo, anything on that topic ? :)

**Reply by imhoffd** [reply]
Are you saying they aren't working in your app? 

The CLI is logging Capacitor plugins, and those are Cordova plugins.

**Reply by daniell0gda** [reply]
I can only ask you to look on [my post](https://github.com/ionic-team/capacitor/discussions/4011#discussioncomment-262358) from before..
It's @ionic-native/backlight which is using cordova plugin - yes, but because it worked in capacitor 2.x I guess it should work now, or at least, in the future.
